streaming/rrdpush.c

```diff
@@ -1396,31 +1427,34 @@ int rrdpush_receiver_thread_spawn(RRDHOST *host, struct web_client *w, char *url
             timezone = value;
         else if(!strcmp(name, "tags"))
             tags = value;
-        else {
-            if(!strcmp(name, "NETDATA_PROTOCOL_VERSION"))
-                stream_flags = LABEL_FLAG_UPDATE_STREAM;
-            else {
-                // An old Netdata slave does not have a compatible streaming protocol, map to something sane.
-                if (!strcmp(name, "NETDATA_SYSTEM_OS_NAME"))
-                    name = "NETDATA_HOST_OS_NAME";
-                else if (!strcmp(name, "NETDATA_SYSTEM_OS_ID"))
-                    name = "NETDATA_HOST_OS_ID";
-                else if (!strcmp(name, "NETDATA_SYSTEM_OS_ID_LIKE"))
-                    name = "NETDATA_HOST_OS_ID_LIKE";
-                else if (!strcmp(name, "NETDATA_SYSTEM_OS_VERSION"))
-                    name = "NETDATA_HOST_OS_VERSION";
-                else if (!strcmp(name, "NETDATA_SYSTEM_OS_VERSION_ID"))
-                    name = "NETDATA_HOST_OS_VERSION_ID";
-                else if (!strcmp(name, "NETDATA_SYSTEM_OS_DETECTION"))
-                    name = "NETDATA_HOST_OS_DETECTION";
-                if (unlikely(rrdhost_set_system_info_variable(system_info, name, value))) {
-                    info("STREAM [receive from [%s]:%s]: request has parameter '%s' = '%s', which is not used.",
+        else if(!strcmp(name, "ver"))
+            stream_version = MIN((uint32_t) strtoul(value, NULL, 0), STREAMING_PROTOCOL_CURRENT_VERSION);
+            // An old Netdata slave does not have a compatible streaming protocol, map to something sane.
+        else if (!strcmp(name, "NETDATA_SYSTEM_OS_NAME"))
+            name = "NETDATA_HOST_OS_NAME";
+        else if (!strcmp(name, "NETDATA_SYSTEM_OS_ID"))
+            name = "NETDATA_HOST_OS_ID";
+        else if (!strcmp(name, "NETDATA_SYSTEM_OS_ID_LIKE"))
+            name = "NETDATA_HOST_OS_ID_LIKE";
+        else if (!strcmp(name, "NETDATA_SYSTEM_OS_VERSION"))
+            name = "NETDATA_HOST_OS_VERSION";
+        else if (!strcmp(name, "NETDATA_SYSTEM_OS_VERSION_ID"))
+            name = "NETDATA_HOST_OS_VERSION_ID";
+        else if (!strcmp(name, "NETDATA_SYSTEM_OS_DETECTION"))
+            name = "NETDATA_HOST_OS_DETECTION";
+        else if(!strcmp(name, "NETDATA_PROTOCOL_VERSION") && stream_version == UINT_MAX) {
+            stream_version = 1;
+        }
+
+        if (unlikely(rrdhost_set_system_info_variable(system_info, name, value)) && stream_version != 1) {
+            info("STREAM [receive from [%s]:%s]: request has parameter '%s' = '%s', which is not used.",
                      w->client_ip, w->client_port, key, value);
```

Comment:
This produces a lot of info messages if the version is not 1 (so as we are currently in version 2) this will always trigger log entries. Also you probably wanted to log `name, value` instead of `key, value`

Reply:
I fixed this with the last commit.